tck/src/main/java/org/eclipse/microprofile/context/tck/ManagedExecutorTest.java

```diff
@@ -1548,23 +1548,20 @@ public void newIncompleteFutureDependentStagesRunWithContext() throws ExecutionE
      */
     @Test
     public void propagateApplicationContext() throws ExecutionException, InterruptedException, TimeoutException {
+        ClassLoader appClassLoader = Thread.currentThread().getContextClassLoader();
+
         ManagedExecutor.Builder builder = ManagedExecutor.builder()
                 .propagated(ThreadContext.APPLICATION)
                 .cleared(ThreadContext.ALL_REMAINING);
 
-        ManagedExecutor executor;
-        try {
-            executor = builder.build();
-        }
-        catch (IllegalStateException x) {
-            return; // Skip test if Application context is not supported.
-        }
+        ManagedExecutor executor = builder.build();
 
         try {
             CompletableFuture<Class<?>> cf = executor.supplyAsync(() -> {
                 try {
                     // load a class from the application
                     ClassLoader loader = Thread.currentThread().getContextClassLoader();
+                    Assert.assertEquals(loader, appClassLoader);
```

Comment:
This still doesn't test anything, does it? This just tests that the new thread has the same TCCL as the original thread, which can easily happen without any propagation happening.

Reply:
Yeah. I was surprised in my tests to realise that all my new threads had a TCCL set to the JDK CL. Not sure who sets them.

Reply:
@Ladicek @FroMage  I added a commit that enhances the test case to ensure the class loader is propagated (and restored afterward).   This switches the test case to execute the completion stage on an unmanaged thread, but with context, where the unmanaged thread has a predetermined class loader that is not the same.  If the context is successfully propagated, then the requesting code's class loader will appear on that thread instead of the predetermined one, and, once it finishes running the completion stage with context, the class loader of that thread will go back to the predetermined one, which the test case also checks for.  Hopefully this will be sufficient.

Reply:
The approach looks good, just added one small but important comment.